revolution_pi/gpio_pin.go

```diff
@@ -55,18 +101,28 @@ func (pin *gpioPin) Set(ctx context.Context, high bool, extra map[string]interfa
 		val = uint8(1)
 	}
 
-	// disable pwm if enabled
+	if !pin.isOutputPWM() && !pin.isOutputWord() {
+		return fmt.Errorf("cannot set pin state, Pin %s is not a digital output pin", pin.Name)
+	}
+
+	// error if using PWM mode
 	if pin.pwmMode {
-		err := pin.disablePwm()
-		if err != nil {
-			return err
-		}
+		return fmt.Errorf("cannot set pin state, Pin %s is configured as PWM", pin.Name)
+	}
+
+	gpioAddress := pin.getGpioAddress()
+	gpioBit := pin.BitPosition
+
+	// if someone used the PWM pin name, get the GPIO address of the pin
+	if !pin.isOutputWord() {
+		gpioBit = uint8((pin.Address-2)%pin.outputOffset) % 8
```

Comment:
Again, most of these calculations are opaque/magical to me. Where can I learn more?

Reply:
we spoke offline showed the docs and went thru the math and found some ways to clean them up